indexer/services/auxo/src/index.ts

```diff
@@ -117,17 +119,37 @@ async function upgradeBazooka(
   });
 
   // Update Lambda function with the new image
-  const response: UpdateFunctionCodeCommandOutput = await lambda.send(
+  await lambda.send(
     new UpdateFunctionCodeCommand({
       FunctionName: BAZOOKA_LAMBDA_FUNCTION_NAME,
       ImageUri: imageUri,
     }),
   );
-  logger.info({
-    at: 'index#upgradeBazooka',
-    message: 'Successfully upgraded bazooka',
-    response,
-  });
+
+  // Polling to check the update status
+  let updateStatus: LastUpdateStatus | string = LastUpdateStatus.InProgress;
+  while (updateStatus === LastUpdateStatus.InProgress) {
+    const statusResponse: GetFunctionCommandOutput = await lambda.send(
+      new GetFunctionCommand({
+        FunctionName: BAZOOKA_LAMBDA_FUNCTION_NAME,
+      }),
+    );
+
+    updateStatus = statusResponse.Configuration!.LastUpdateStatus!;
+    if (updateStatus === LastUpdateStatus.Successful) {
+      logger.info({
+        at: 'index#upgradeBazooka',
+        message: 'Successfully upgraded bazooka',
+        response: statusResponse,
+      });
+      return;
+    } else if (updateStatus === LastUpdateStatus.Failed) {
+      throw new Error('Failed to upgrade bazooka');
+    }
+
+    // Wait for 5s before checking again
+    await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
nit: 1 minute timeout + exit the function?
Non-blocking

Reply:
I added a 2 min timeout. Updating lambda image can be slow in my experience.